File: braingenerators/src/main/java/com/graphbrain/braingenerators/Formatting.java
package com.graphbrain.braingenerators;

import java.util.regex.Matcher;
import java.util.regex.Pattern;

public class Formatting {

    private static Pattern listRegex = Pattern.compile("List(s?)\\sof\\s");
    private static Pattern wordRegex = Pattern.compile("[a-z]+|[A-Z][a-z]+");
    //val WikiLinkRegex="""\[\[([^\]]*)\]\]""".r
    //val CategoryRegex="""category\:.*""".r
    //val IrregularTitleRegex=""".*\:.*""".r
	
	public static String normalizeWikiTitle(String rawTitle) {
   		String title = rawTitle.trim();
    	
    	String norm_title = "";
    	if (title.length() > 0) {
            norm_title += ("" + title.charAt(0)).toUpperCase();
        }
    
    	if (title.length() > 1) {
            norm_title += title.substring(1, title.length());
    	}
    	norm_title = norm_title.replace("_", " ");
      
    	return norm_title;
    }

    public static boolean isList(String titleString) {
        return listRegex.matcher(titleString).find();
    }

    public static String separateWords(String stringToSep) {
        Matcher words = wordRegex.matcher(stringToSep);
        //val words = wordRegex.findAllIn(stringToSep)

        String returnString = "";
        while(words.find()) {
            returnString += " " + words.group().toLowerCase().trim();
        }
        return returnString;
    }

    /*
  def stripLink(link:String):String =
    link.replaceAll("""(\[\[)""", "").replaceAll("""\]\]""", "")

  def wikiLink(title:String):String =
    title.replace(" ", "_")

  def isCategory(titleString:String):Boolean =
    CategoryRegex.findAllIn(titleString).hasNext



  def isIrregular(titleString:String):Boolean =
    IrregularTitleRegex.findAllIn(titleString).hasNext
  */
  
}

File: old/braingenerators/src/main/scala/com/graphbrain/braingenerators/Formatting.scala
package com.graphbrain.braingenerators

object Formatting {

  val ListRegex = """List(s?)\sof\s""".r
  val WikiLinkRegex="""\[\[([^\]]*)\]\]""".r
  val CategoryRegex="""category\:.*""".r
  val IrregularTitleRegex=""".*\:.*""".r
	
	def normalizeWikiTitle(rawTitle:String):String={
   		val title=rawTitle.trim.stripLineEnd
    	
    	var norm_title = ""
    	if (title.length > 0) {
        norm_title += title(0).toString.toUpperCase
      }
    
    	if (title.length > 1) {
        norm_title += title.substring(1, title.length)
    	}
    	norm_title = norm_title.replace("_", " ")
      
    	norm_title
  }

  def stripLink(link:String):String =
    link.replaceAll("""(\[\[)""", "").replaceAll("""\]\]""", "")

  def wikiLink(title:String):String =
    title.replace(" ", "_")

  def isCategory(titleString:String):Boolean =
    CategoryRegex.findAllIn(titleString).hasNext

  def isList(titleString:String):Boolean =
    ListRegex.findAllIn(titleString).hasNext

  def isIrregular(titleString:String):Boolean =
    IrregularTitleRegex.findAllIn(titleString).hasNext

  def separateWords(stringToSep:String):String = {
     val wordRegex="""[a-z]+|[A-Z][a-z]+""".r
     val words=wordRegex.findAllIn(stringToSep)
     
     var returnString=""
     while(words.hasNext) {
       returnString += " " + words.next().toLowerCase.trim
     }
     returnString
  }

  
}